File: PrismEditor/Assets/scripts/Python/RandomColor.py
import random
from Prism import Behaviour, MeshRendererComponent, Time, Log
from Prism.Math import Vector3


class RandomColor(Behaviour):
    def OnCreate(self):
        self.random = random.Random()
        meshComponent = self.GetComponent(MeshRendererComponent)
        self.material = meshComponent.GetMaterial(0)
        self.GenerateColor()
        self._timer = 0.0

    def OnUpdate(self):
        self._timer += Time.DeltaTime
        if self._timer >= 1.0:
            self._timer = 0.0
            self.GenerateColor()
        pass

    def GenerateColor(self):
        if not self.Enabled: return
        r = self.random.random()
        g = self.random.random()
        b = self.random.random()
        self.material.SetVector3("u_AlbedoColor", Vector3(r, g, b))

File: ExampleApp/src/RandomColor.cs
using Prism;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Example
{
    class RandomColor : Behaviour
    {
        public float Timer = 0;
        private Random random = new Random();
        private Material material;
        private void OnCreate()
        {
            MeshRendererComponent meshComponent = GetComponent<MeshRendererComponent>();
            material = meshComponent.Material;
            GenerateColor();
            Timer = 0;
        }
        private void OnUpdate()
        {
            Timer += Time.DeltaTime;
            if (Timer > 1.5 )
            {
                GenerateColor();
                Timer = 0;
            }
        }
        public void GenerateColor()
        {
            if (!Enabled) return;
            float r = (float)random.NextDouble();
            float g = (float)random.NextDouble();
            float b = (float)random.NextDouble();
            material.SetVector3("u_AlbedoColor", new Vector3(r, g, b));
        }
    }
}
